classes/TwitterClient.php

```diff
@@ -77,26 +82,26 @@ public function getLatestTweet($params)
             'params' => array(
                 'count'            => $params["tweet-limit"],
                 'screen_name'      => $userData['screen_name'],
-                'exclude_replies'  => (!isset($params["exclude-replies"]) &&
-                                        $params["exclude-replies"] == 'No' ? false : true) 
+                'exclude_replies'  => (!isset($params["exclude-replies"]) && $params["exclude-replies"] == 'No' ? false : true),
+                'cache-duration'   => (!isset($params["cache-duration"]) ? 10 : $params["cache-duration"])
             )
         ));
 
         if ($code <> 200)
         {
              throw new ApplicationException('Error requesting Twitter API: '.$obj->client->response['error']);
         }
-           
+
         $result = json_decode($obj->client->response['response'], true);
-        foreach ($result as &$message) 
+        foreach ($result as &$message)
         {
             $text = $message['text'];
             $text = preg_replace('/\@\w+/', '<span class="name">$0</span>', $text);
             $text = preg_replace('/\#\w+/', '<span class="tag">$0</span>', $text);
             $message['text_processed'] = $obj->urlsToLinks($text);
         }
 
-        Cache::put($cacheKey, serialize($result), 2);
+        Cache::put($cacheKey, serialize($result), $params["cache-duration"]);
```

Comment:
```suggestion
        Cache::put($cacheKey, serialize($result), now()->addMinutes($params["cache-duration"]));
```
The cache interface uses seconds as the interval unit, it's better to use a carbon instance for code clarity here.